android/src/main/java/com/swmansion/gesturehandler/core/NativeViewGestureHandler.kt

```diff
@@ -44,7 +45,13 @@ class NativeViewGestureHandler : GestureHandler<NativeViewGestureHandler>() {
 
   override fun shouldRecognizeSimultaneously(handler: GestureHandler<*>): Boolean {
     // if the gesture is marked by user as simultaneous with other or the hook return true
-    if (super.shouldRecognizeSimultaneously(handler) || hook.shouldRecognizeSimultaneously(handler)) {
+    val hookCheckResult = hook.shouldRecognizeSimultaneously(handler)
+
+    if (hookCheckResult != null) {
+      return hookCheckResult
+    }
```

Comment:
```suggestion
    hook.shouldRecognizeSimultaneously(handler)?.let {
      return@shouldRecognizeSimultaneously it
    }
```

Doesn't something like this work?

Reply:
Seems like it does 😅 

Done in [1852e67](https://github.com/software-mansion/react-native-gesture-handler/pull/3202/commits/1852e67f5cf149b59a0f293225c0406b74ff6cc2)